pineappl/src/grid.rs

```diff
@@ -1389,10 +1389,22 @@ impl Grid {
     }
 
     fn symmetrize_channels(&mut self) {
-        if self.key_values().map_or(false, |map| {
-            map["initial_state_1"] != map["initial_state_2"]
-        }) {
-            return;
+        let map = self.key_values();
+        if map.is_some() {
+            let map = map.unwrap();
+            if map.contains_key("convolution_particle_1")
+                && map.contains_key("convolution_type_1")
+                && map.contains_key("convolution_particle_2")
+                && map.contains_key("convolution_type_2")
+            {
+                if map["convolution_particle_1"] != map["convolution_particle_2"]
+                    || map["convolution_type_1"] != map["convolution_type_2"]
+                {
```

Comment:
Similar to the `if`-statement above you can write these two `if`s using `match`:
```rust
match (map.get("convolution_particle_1"), map.get("convolution_particle_2"), map.get("convolution_type_1"), map.get("convolution_type_1")) {
    (Some(convolution_particle_1), Some(convolution_particle_2), Some(convolution_type_1), Some(convolution_type_2)) => (convolution_particle_1 != convolution_particle_2) || (convolution_type_1 != convolution_type_2),
    (None, None, None, None) => {},
    _ => {
        // TODO: if only some of the metadata is set, we should consider this an error
        todo!();
    }
}
```

```

Reply:
done, please let me know how to impkement this error